File: Biblioteca/Materiais/Strings/chaves_colchetes_parenteses.py
# Descricao: Funcao que verifica se uma string contendo chaves, colchetes e parenteses esta balanceada.
def balanced(string) -> bool:
   stack = []
  
   for i in string:
       if i in '([{': stack.append(i)
          
       elif i in ')]}':
           if (not stack) or ((stack[-1],i) not in [('(',')'), ('[',']'), ('{','}')]):
               return False
           else:
               stack.pop()
      
   return not stack

File: Biblioteca/Materiais/Strings/chaves_colchetes_parenteses.cpp
// Description: Verifica se s tem uma sequência valida de {}, [] e ()
// Complexidade: O(n)
bool brackets(string s) {
    stack<char> st;

    for (char c : s) {
        if (c == '(' || c == '[' || c == '{') {
            st.push(c);
        } else {
            if (st.empty()) return false;
            if (c == ')' and st.top() != '(') return false;
            if (c == ']' and st.top() != '[') return false;
            if (c == '}' and st.top() != '{') return false;
            st.pop();
        }
    }

    return st.empty();
}